tests/lib/rules/array-bracket-spacing.js

```diff
@@ -100,14 +91,6 @@ ruleTester.run("array-bracket-spacing", rule, {
         { code: "[ x, { y, z }] = arr;", options: ["always", { objectsInArrays: false }], parserOptions: { ecmaVersion: 6 } },
 
         // never
-        { code: "obj[foo]", options: ["never"] },
-        { code: "obj['foo']", options: ["never"] },
-        { code: "obj['foo' + 'bar']", options: ["never"] },
-        { code: "obj['foo'+'bar']", options: ["never"] },
-        { code: "obj[obj2[foo]]", options: ["never"] },
-        { code: "obj.map(function(item) { return [\n1,\n2,\n3,\n4\n]; })", options: ["never"] },
-        { code: "obj['map'](function(item) { return [\n1,\n2,\n3,\n4\n]; })", options: ["never"] },
-        { code: "obj['for' + 'Each'](function(item) { return [\n1,\n2,\n3,\n4\n]; })", options: ["never"] },
```

Comment:
I think we need simpler array bracket cases for these as well.

I may close this PR and make some more of the updates together.